### PR TITLE
Powershell changes for service endpointpolicies

### PR DESCRIPTION
**Pull request by chandrasekarsrinivasan**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 1]
```diff
@@ -57,11 +79,20 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.KeyVault.Core.1.0.0\lib\net40\Microsoft.Azure.KeyVault.Core.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=15.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.15.1.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=19.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=AMD64">
```
Comment (chandrasekarsrinivasan) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: Remove the redundant references

File: src/ResourceManager/Network/Commands.Network/Models/PSSubnet.cs [hunk 2]
```diff
@@ -74,5 +76,11 @@ public string ServiceEndpointText
         {
             get { return JsonConvert.SerializeObject(ServiceEndpoints, Formatting.Indented, new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore }); }
         }
+
+        [JsonIgnore]
+        public string ServiceEndpointPoliciesText
```
Comment (chandrasekarsrinivasan) [inline src/ResourceManager/Network/Commands.Network/Models/PSSubnet.cs]: Add text for definitions also

**Comment by praries880** [comment]
@chandrasekarsrinivasan  you have more merge conflicts that need resolution...

File: src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/GetAzureServiceEndpointPolicyCommand.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Network
+{
+    using Microsoft.Azure.Commands.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.Get, "AzureRmServiceEndpointPolicy"), OutputType(typeof(PSServiceEndpointPolicy))]
+    public class GetAzureServiceEndpointPolicyCommand : ServiceEndpointPolicyBaseCmdlet
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/GetAzureServiceEndpointPolicyCommand.cs]: Is there no list feature when getting an endpoint policy? Like if the user only specified
* RG only : the user gets all the policiess in the RG
* No RG and No name : All the policies in the subscription.

File: src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/RemoveAzureServiceEndpointPolicyCommand.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Network
+{
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+    [Cmdlet(VerbsCommon.Remove, "AzureRmServiceEndpointPolicy", SupportsShouldProcess = true)]
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/RemoveAzureServiceEndpointPolicyCommand.cs]: Can you add support for REsourceId and InputObject to support [piping guidelines](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/azure-powershell-design-guidelines.md#piping-guidelines)?

File: src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/SetAzureServiceEndpointPolicyCommand.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Network
+{
+    using Microsoft.Azure.Commands.Network.Models;
+    using System;
+    using System.Management.Automation;
+    using MNM = Microsoft.Azure.Management.Network.Models;
+
+    [Cmdlet(VerbsCommon.Set, "AzureRmServiceEndpointPolicy"), OutputType(typeof(PSServiceEndpointPolicy))]
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/SetAzureServiceEndpointPolicyCommand.cs]: This cmdlet might benefit from using [ShouldProcess](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/azure-powershell-design-guidelines.md#shouldprocess)

**Comment by markcowl** [comment]
@chandrasekarsrinivasan Code complete is Today (8/6).  If you want this change in the next release, please fix merge conflicts and address review comments

**Comment by chandrasekarsrinivasan** [comment]
I'm targeting 8/24 release.

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 6]
```diff
@@ -57,11 +57,8 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.KeyVault.Core.1.0.0\lib\net40\Microsoft.Azure.KeyVault.Core.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=15.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: why is this being removed?

**Comment by praries880** [comment]
@chandrasekarsrinivasan  merge conflicts need to be resolved as well

File: src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/SetAzureServiceEndpointPolicyCommand.cs (hunk 5 above)
Comment (chandrasekarsrinivasan) [reply src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/SetAzureServiceEndpointPolicyCommand.cs]: Fixed

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 7]
```diff
@@ -57,11 +57,11 @@
...
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.15.1.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=19.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=AMD64">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.21.1.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
     </Reference>
     <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL" />
+    <Reference Include="System.Runtime" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: why is this change needed?

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj (hunk 1 above)
Comment (praries880) [reply src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: so there's a discrepancy here :
```
    <Reference Include="Microsoft.Azure.Management.Network, Version=19.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=AMD64"> 
    <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.21.1.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>  
      <Private>True</Private> 
    </Reference>
```
can you make the version number same in both the places?

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj (hunk 7 above)
Comment (praries880) [reply src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: revert the client runtime and Newtonsoft related changes here... the libraries are automatically puiled in..

File: src/ResourceManager/Network/Commands.Network/packages.config [hunk 8]
```diff
@@ -2,5 +2,8 @@
 <packages>
   <package id="AutoMapper" version="6.0.2" targetFramework="net452" />
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Network" version="19.1.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Network" version="20.2.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.12" targetFramework="net452" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/packages.config]: remove the slient runtime and newtonsoft changes here

**Comment by praries880** [comment]
@chandrasekarsrinivasan Issues to fix in the help files :


Target | Description
-- | --
Add-AzureRmVirtualNetworkSubnetConfig.md | Trailing yaml string found in description for parameter   ServiceEndpointPolicies. Please move the trailing yaml string to a line by   itself.
Get-AzureRmServiceEndpointPolicy.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/get-azurermserviceendpointpolicy
Get-AzureRmServiceEndpointPolicy.md | No examples found
Get-AzureRmServiceEndpointPolicyDefinition.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/get-azurermserviceendpointpolicydefinition
Get-AzureRmServiceEndpointPolicyDefinition.md | No examples found
New-AzureRmServiceEndpointPolicy.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/new-azurermserviceendpointpolicy
New-AzureRmServiceEndpointPolicy.md | No examples found
New-AzureRmServiceEndpointPolicyDefinition.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/new-azurermserviceendpointpolicydefinition
New-AzureRmServiceEndpointPolicyDefinition.md | No examples found
Remove-AzureRmServiceEndpointPolicy.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/remove-azurermserviceendpointpolicy
Remove-AzureRmServiceEndpointPolicy.md | No examples found
Remove-AzureRmServiceEndpointPolicyDefinition.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/remove-azurermserviceendpointpolicydefinition
Remove-AzureRmServiceEndpointPolicyDefinition.md | No examples found
Set-AzureRmServiceEndpointPolicy.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/set-azurermserviceendpointpolicy
Set-AzureRmServiceEndpointPolicy.md | No examples found
Set-AzureRmServiceEndpointPolicyDefinition.md | No entry was found for the online version field in the header. The   corresponding URL should be the following:   https://docs.microsoft.com/en-us/powershell/module/azurerm.network/set-azurermserviceendpointpolicydefinition
Set-AzureRmServiceEndpointPolicyDefinition.md | No examples found

**Comment by praries880** [comment]
@MikhailTryakhov  can you review this PR as well? :)

File: src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/ServiceEndpointPolicyDefinition/AddAzureServiceEndpointPolicyDefinitionCommand.cs [hunk 9]
```diff
@@ -0,0 +1,65 @@
+﻿// ----------------------------------------------------------------------------------
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/ServiceEndpointPolicyDefinition/AddAzureServiceEndpointPolicyDefinitionCommand.cs]: @chandrasekarsrinivasan 
help file missing for this command,

"Assembly","HelpFile","Target","Severity","ProblemId","Description","Remediation"
"","","Microsoft.Azure.Commands.Network.AddAzureServiceEndpointPolicyDefinitionCommand","1","6050","Help missing for cmdlet Add-AzureRmServiceEndpointPolicyDefinition implemented by class Microsoft.Azure.Commands.Network.AddAzureServiceEndpointPolicyDefinitionCommand","Add Help record for cmdlet Add-AzureRmServiceEndpointPolicyDefinition to help file."

File: src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/NewAzureServiceEndpointPolicyCommand.cs [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Parameter(
+         Mandatory = true,
+         ValueFromPipelineByPropertyName = true,
+         HelpMessage = "location.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Location { get; set; }
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ServiceEndpointPolicy/NewAzureServiceEndpointPolicyCommand.cs]: use the [LocationCompleter](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/azure-powershell-developer-guide.md#location-completer)

File: src/ResourceManager/Network/Commands.Network/packages.config [hunk 11]
```diff
@@ -2,5 +2,8 @@
 <packages>
   <package id="AutoMapper" version="6.0.2" targetFramework="net452" />
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Network" version="19.1.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Network" version="20.2.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.11" targetFramework="net452" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/packages.config]: revert this change

File: src/ResourceManager/Network/Commands.Network/packages.config [hunk 12]
```diff
@@ -2,5 +2,8 @@
...
+  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.12" targetFramework="net452" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/packages.config]: revert this change

File: src/ResourceManager/Network/Commands.Network/packages.config [hunk 13]
```diff
@@ -2,5 +2,8 @@
...
+  <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net452" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/packages.config]: revert this change

File: src/ResourceManager/Network/Commands.Network/packages.config (hunk 12 above)
Comment (chandrasekarsrinivasan) [reply src/ResourceManager/Network/Commands.Network/packages.config]: fixed

File: src/ResourceManager/Network/Commands.Network/packages.config (hunk 11 above)
Comment (chandrasekarsrinivasan) [reply src/ResourceManager/Network/Commands.Network/packages.config]: fixed

File: src/ResourceManager/Network/Commands.Network/packages.config (hunk 13 above)
Comment (chandrasekarsrinivasan) [reply src/ResourceManager/Network/Commands.Network/packages.config]: fixed

File: src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1 [hunk 14]
```diff
@@ -335,6 +335,7 @@ CmdletsToExport = 'Add-AzureRmApplicationGatewayAuthenticationCertificate',
                'Remove-AzureRmServiceEndpointPolicyDefinition', 
                'Get-AzureRmServiceEndpointPolicyDefinition',
                'Set-AzureRmServiceEndpointPolicyDefinition',
+               'Add-AzureRmServiceEndpointPolicyDefinition',
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1]: is this change present in the netcore psd1 as well? if not, do kindly add it there as well...

Comment (chandrasekarsrinivasan) [reply src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1]: Fixed.

**Comment by praries880** [comment]
@chandrasekarsrinivasan Need to fix these issues with the cmdlets :

AssemblyFileName | ClassName | Target | Severity | ProblemId | Description | Remediation
-- | -- | -- | -- | -- | -- | --
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.AddAzureServiceEndpointPolicyDefinitionCommand | Add-AzureRmServiceEndpointPolicyDefinition | 1 | 8100 | Add-AzureRmServiceEndpointPolicyDefinition Does not support ShouldProcess   but the cmdlet verb Add indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.GetAzureServiceEndpointPolicyDefinitionCommand | Get-AzureRmServiceEndpointPolicyDefinition | 1 | 8410 | Parameter DefaultRules of cmdlet   Get-AzureRmServiceEndpointPolicyDefinition does not follow the enforced   naming convention of using a singular noun for a parameter name. | Consider using a singular noun for the parameter name.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.NewAzureServiceEndpointPolicyDefinitionCommand | New-AzureRmServiceEndpointPolicyDefinition | 1 | 8100 | New-AzureRmServiceEndpointPolicyDefinition Does not support ShouldProcess   but the cmdlet verb New indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzureServiceEndpointPolicyDefinitionCommand | Remove-AzureRmServiceEndpointPolicyDefinition | 1 | 8100 | Remove-AzureRmServiceEndpointPolicyDefinition Does not support   ShouldProcess but the cmdlet verb Remove indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzureServiceEndpointPolicyDefinitionCommand | Remove-AzureRmServiceEndpointPolicyDefinition | 1 | 8510 | Cmdlet 'Remove-AzureRmServiceEndpointPolicyDefinition' has multiple   parameter sets, but no defined default parameter set. | Define a default parameter set in the cmdlet attribute.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.SetAzureServiceEndpointPolicyDefinitionCommand | Set-AzureRmServiceEndpointPolicyDefinition | 1 | 8100 | Set-AzureRmServiceEndpointPolicyDefinition Does not support ShouldProcess   but the cmdlet verb Set indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.GetAzureServiceEndpointPolicyCommand | Get-AzureRmServiceEndpointPolicy | 1 | 8510 | Cmdlet 'Get-AzureRmServiceEndpointPolicy' has multiple parameter sets,   but no defined default parameter set. | Define a default parameter set in the cmdlet attribute.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzureServiceEndpointPolicyCommand | Remove-AzureRmServiceEndpointPolicy | 1 | 8600 | Cmdlet 'Remove-AzureRmServiceEndpointPolicy' has no defined output type. | Add an OutputType attribute that declares the type of the object(s)   returned by this cmdlet. If this cmdlet returns no output, please set the   output type to 'bool' and make sure to implement the 'PassThru' parameter.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzureServiceEndpointPolicyCommand | Remove-AzureRmServiceEndpointPolicy | 1 | 8510 | Cmdlet 'Remove-AzureRmServiceEndpointPolicy' has multiple parameter sets,   but no defined default parameter set. | Define a default parameter set in the cmdlet attribute.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.SetAzureServiceEndpointPolicyCommand | Set-AzureRmServiceEndpointPolicy | 1 | 8100 | Set-AzureRmServiceEndpointPolicy Does not support ShouldProcess but the   cmdlet verb Set indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.SetAzureVirtualNetworkSubnetConfigCommand | Set-AzureRmVirtualNetworkSubnetConfig | 1 | 8410 | Parameter ServiceEndpointPolicies of cmdlet   Set-AzureRmVirtualNetworkSubnetConfig does not follow the enforced naming   convention of using a singular noun for a parameter name. | Consider using a singular noun for the parameter name.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.AddAzureVirtualNetworkSubnetConfigCommand | Add-AzureRmVirtualNetworkSubnetConfig | 1 | 8410 | Parameter ServiceEndpointPolicies of cmdlet   Add-AzureRmVirtualNetworkSubnetConfig does not follow the enforced naming   convention of using a singular noun for a parameter name. | Consider using a singular noun for the parameter name.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.NewAzureVirtualNetworkSubnetConfigCommand | New-AzureRmVirtualNetworkSubnetConfig | 1 | 8410 | Parameter ServiceEndpointPolicies of cmdlet   New-AzureRmVirtualNetworkSubnetConfig does not follow the enforced naming   convention of using a singular noun for a parameter name. | Consider using a singular noun for the parameter name.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.AddAzureServiceEndpointPolicyDefinitionCommand | Add-AzureRmServiceEndpointPolicyDefinition | 1 | 8100 | Add-AzureRmServiceEndpointPolicyDefinition Does not support ShouldProcess   but the cmdlet verb Add indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.GetAzureServiceEndpointPolicyDefinitionCommand | Get-AzureRmServiceEndpointPolicyDefinition | 1 | 8410 | Parameter DefaultRules of cmdlet   Get-AzureRmServiceEndpointPolicyDefinition does not follow the enforced   naming convention of using a singular noun for a parameter name. | Consider using a singular noun for the parameter name.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.NewAzureServiceEndpointPolicyDefinitionCommand | New-AzureRmServiceEndpointPolicyDefinition | 1 | 8100 | New-AzureRmServiceEndpointPolicyDefinition Does not support ShouldProcess   but the cmdlet verb New indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzureServiceEndpointPolicyDefinitionCommand | Remove-AzureRmServiceEndpointPolicyDefinition | 1 | 8100 | Remove-AzureRmServiceEndpointPolicyDefinition Does not support   ShouldProcess but the cmdlet verb Remove indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzureServiceEndpointPolicyDefinitionCommand | Remove-AzureRmServiceEndpointPolicyDefinition | 1 | 8510 | Cmdlet 'Remove-AzureRmServiceEndpointPolicyDefinition' has multiple   parameter sets, but no defined default parameter set. | Define a default parameter set in the cmdlet attribute.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.SetAzureServiceEndpointPolicyDefinitionCommand | Set-AzureRmServiceEndpointPolicyDefinition | 1 | 8100 | Set-AzureRmServiceEndpointPolicyDefinition Does not support ShouldProcess   but the cmdlet verb Set indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.GetAzureServiceEndpointPolicyCommand | Get-AzureRmServiceEndpointPolicy | 1 | 8510 | Cmdlet 'Get-AzureRmServiceEndpointPolicy' has multiple parameter sets,   but no defined default parameter set. | Define a default parameter set in the cmdlet attribute.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzureServiceEndpointPolicyCommand | Remove-AzureRmServiceEndpointPolicy | 1 | 8600 | Cmdlet 'Remove-AzureRmServiceEndpointPolicy' has no defined output type. | Add an OutputType attribute that declares the type of the object(s)   returned by this cmdlet. If this cmdlet returns no output, please set the   output type to 'bool' and make sure to implement the 'PassThru' parameter.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.RemoveAzureServiceEndpointPolicyCommand | Remove-AzureRmServiceEndpointPolicy | 1 | 8510 | Cmdlet 'Remove-AzureRmServiceEndpointPolicy' has multiple parameter sets,   but no defined default parameter set. | Define a default parameter set in the cmdlet attribute.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.SetAzureServiceEndpointPolicyCommand | Set-AzureRmServiceEndpointPolicy | 1 | 8100 | Set-AzureRmServiceEndpointPolicy Does not support ShouldProcess but the   cmdlet verb Set indicates that it should. | Determine if the cmdlet should implement ShouldProcess and if so   determine if it should implement Force / ShouldContinue
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.SetAzureVirtualNetworkSubnetConfigCommand | Set-AzureRmVirtualNetworkSubnetConfig | 1 | 8410 | Parameter ServiceEndpointPolicies of cmdlet   Set-AzureRmVirtualNetworkSubnetConfig does not follow the enforced naming   convention of using a singular noun for a parameter name. | Consider using a singular noun for the parameter name.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.AddAzureVirtualNetworkSubnetConfigCommand | Add-AzureRmVirtualNetworkSubnetConfig | 1 | 8410 | Parameter ServiceEndpointPolicies of cmdlet   Add-AzureRmVirtualNetworkSubnetConfig does not follow the enforced naming   convention of using a singular noun for a parameter name. | Consider using a singular noun for the parameter name.
c:\workspace\powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll | Microsoft.Azure.Commands.Network.NewAzureVirtualNetworkSubnetConfigCommand | New-AzureRmVirtualNetworkSubnetConfig | 1 | 8410 | Parameter ServiceEndpointPolicies of cmdlet   New-AzureRmVirtualNetworkSubnetConfig does not follow the enforced naming   convention of using a singular noun for a parameter name. | Consider using a singular noun for the parameter name.

**Comment by praries880** [comment]
@chandrasekarsrinivasan The netcore build is breaking because the test code cant find the new network SDK : 
https://travis-ci.org/Azure/azure-powershell/builds/414278002#L1542

**Comment by praries880** [comment]
@chandrasekarsrinivasan  Kindly add a change log entry [here](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/ChangeLog.md#current-release).

**Comment by praries880** [comment]
job to re-record tests : https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-network-test/45/

**Comment by sphibbs** [comment]
@chandrasekarsrinivasan please resolve conflicts

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 15]
```diff
@@ -64,8 +64,8 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Authorization.2.0.0\lib\net40\Microsoft.Azure.Management.Authorization.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=19.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.20.1.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=20.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (MikhailTryakhov) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: Different package version in the config (20.2.0 and 20.1.0)

File: src/ResourceManager/Network/Commands.Network.Test/packages.config [hunk 16]
```diff
@@ -8,7 +8,7 @@
   <package id="Microsoft.Azure.Insights" version="0.13.1-preview" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Authorization" version="2.0.0" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.Compute" version="21.0.0" targetFramework="net452" />
-  <package id="Microsoft.Azure.Management.Network" version="19.1.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Network" version="20.1.0-preview" targetFramework="net452" />
```
Comment (MikhailTryakhov) [inline src/ResourceManager/Network/Commands.Network.Test/packages.config]: different versions

**Comment by sphibbs** [comment]
Merging per email discussions. PR 6719 has fixes for the test issues. As stated in email, we will not merge anything from this branch into the preview branch unless ALL tests are passing.
